x-pack/plugins/observability_solution/observability_onboarding/public/application/quickstart_flows/firehose/index.tsx

```diff
@@ -52,13 +130,68 @@ export function FirehosePanel() {
             </>
           )}
           {status === FETCH_STATUS.SUCCESS && data !== undefined && (
-            <CreateStackCommandSnippet
-              templateUrl={data.templateUrl}
-              encodedApiKey={data.apiKeyEncoded}
-              onboardingId={data.onboardingId}
-              elasticsearchUrl={data.elasticsearchUrl}
-              isCopyPrimaryAction={!isVisualizeStepActive}
-            />
+            <>
+              <EuiButtonGroup
+                legend={i18n.translate(
+                  'xpack.observability_onboarding.firehosePanel.createStackOptionsLegend',
+                  {
+                    defaultMessage:
+                      'Select one an a preferred option to create a CloudFormation stack',
+                  }
+                )}
+                type="single"
+                buttonSize="m"
+                idSelected={selectedOptionId}
+                onChange={onOptionChange}
+                options={OPTIONS}
+              />
+
+              <EuiSpacer size="l" />
+
+              {selectedOptionId === CreateStackOption.AWS_CONSOLE_UI && (
+                <CreateStackInAWSConsole
+                  templateUrl={data.templateUrl}
+                  encodedApiKey={data.apiKeyEncoded}
+                  elasticsearchUrl={data.elasticsearchUrl}
+                  isPrimaryAction={!isVisualizeStepActive}
+                />
+              )}
+
+              {selectedOptionId === CreateStackOption.AWS_CLI && (
+                <CreateStackCommandSnippet
+                  templateUrl={data.templateUrl}
+                  encodedApiKey={data.apiKeyEncoded}
+                  elasticsearchUrl={data.elasticsearchUrl}
+                  isCopyPrimaryAction={!isVisualizeStepActive}
+                />
+              )}
+
+              <EuiHorizontalRule />
+
+              <EuiText>
+                <p>
+                  <FormattedMessage
+                    id="xpack.observability_onboarding.firehosePanel.DownloadCloudFormationTemplateCaption"
+                    defaultMessage="Download and edit the CloudFormation Template if necessary:"
+                  />
+                </p>
+              </EuiText>
```

Comment:
Isn't this its own flow? I'm not clear how downloading and editing the raw CloudFormation template ties into the AWS console or the AWS CLI steps. I think we need to explain that. 

What would be a scenario when a user would want to do this?

Reply:
You're right, downloading the template is not part of either flow, but there was a requirement from @akhileshpok to have a link to the template based on some feedback he got. We could have this link in the docs though... I'll talk to Akhilesh and Christian about this.

Reply:
Christian suggested to use `EuiButtonEmpty`, I think it looks less distracting now and feels more like an auxiliary thing. wdyt?

![CleanShot 2024-09-16 at 16 58 46@2x](https://github.com/user-attachments/assets/1d8b5165-1750-4474-86e3-ed02d0364d8e)

Reply:
The use-case here is for people who want to use our template with their infrastructure-as-code tooling. They can download the template, modify it and use as part of their setup. You're right that it still indeed feels like an extra (or optional) step in the flow, while in reality it's a way to break out of the flow.

I'll bring this back to Akhilesh and Christian to discuss.

Reply:
Yeh if that's the case then it would make more sense to have a "Custom tooling" tab and add instructions there rather than mixing it in with the other flows

Reply:
Now moved the link to the step description as discussed during yesterday's sync
![CleanShot 2024-09-18 at 11 52 05@2x](https://github.com/user-attachments/assets/8722e54f-bd25-4d77-8dd5-fa7ed0a759dd)